examples/armv4t/gdb/host_io.rs

```diff
@@ -28,14 +44,55 @@ impl target::ext::host_io::HostIoOpen for Emu {
     fn open(
         &mut self,
         filename: &[u8],
-        _flags: HostIoOpenFlags,
-        _mode: HostIoOpenMode,
+        flags: HostIoOpenFlags,
+        mode: HostIoOpenMode,
     ) -> HostIoResult<u32, Self> {
         // Support `info proc mappings` command
         if filename == b"/proc/1/maps" {
```

Comment:
Oh shit, it's really ugly.
I think the handle of `/proc/` could be stripped away now. Let the user consider about it.

Reply:
I mean, it's not _too_ ugly, and if you're going to provide filesystem access, you'll want to trap the proc path for correctness. Otherwise, the armv4t example will return bogus data when running `info proc mappings`, which isn't good.

Reply:
(I've expanded on my thoughts regarding `/proc/` in a top-level response)